src/CommunityToolkit.Aspire.Hosting.SqlDatabaseProjects/SqlProjectBuilderExtensions.cs

```diff
@@ -198,35 +198,70 @@ internal static IResourceBuilder<TResource> InternalWithReference<TResource>(thi
         {
             builder.ApplicationBuilder.Eventing.Subscribe<ResourceReadyEvent>(target.Resource, async (resourceReady, ct) =>
             {
-                await RunPublish(builder, target, targetDatabaseName, resourceReady.Services, ct);
+                if (builder.Resource.HasAnnotationOfType<ExplicitStartupAnnotation>())
+                {
+                    await MarkNotStarted(builder, resourceReady.Services);
+                }
+                else
+                {
+                    await RunPublish(builder, target, targetDatabaseName, resourceReady.Services, ct);
+                }
             });
         }
-        else 
+        else
```

Comment:
Can't we combine these two if-branches into one? I believe the only real difference is the event we're subscribing to?